scripts/utils/generate_vif/generate_vif.py

```diff
@@ -0,0 +1,243 @@
+#!/usr/bin/env python3
+
+# Copyright (c) 2022 The Chromium OS Authors
+# SPDX-License-Identifier: Apache-2.0
+
+import argparse
+import os
+import sys
+import xml.etree.ElementTree as ET
+
+import constants
+
+sys.path.insert(0, os.path.join(
+    os.path.dirname(os.path.dirname(os.path.dirname(__file__))), 'dts',
+    'python-devicetree', 'src'))
+
+from devicetree import edtlib
+
+
+def main():
+  args = parse_args()
+  try:
+    edt = edtlib.EDT(args.dts, args.bindings_dirs)
+  except edtlib.EDTError as e:
+    sys.exit(f"devicetree error: {e}")
+  root = get_root()
+  add_elements_from_dict(constants.VIF_SPEC_ELEMENTS, root)
+  add_xml_element(root, constants.MODEL_PART_NUMBER, args.board)
+  xml_mem = add_xml_element(root, constants.COMPONENT)
+  for node in edt.compat2nodes[args.compatible]:
+    parse_node(xml_mem, node)
+  tree = ET.ElementTree(root)
+  tree.write(args.vif_out, xml_declaration=True,
+             encoding=constants.XML_ENCODING)
+
+
+def is_vif_element(name):
+  if name in constants.VIF_ELEMENTS:
+    return True
+  return False
+
+
+def get_vif_element_name(string):
+  return constants.XML_ELEMENT_NAME_PREFIX + ":" + constants.DT_VIF_ELEMENTS_DICT.get(
+      string, string)
+
+
+def get_root():
+  root = ET.Element(get_vif_element_name(constants.XML_ROOT_ELEMENT_NAME))
+  add_attributes_from_dict(constants.XML_NAMESPACE_ATTRIBUTES, root)
+  return root
+
+
+def add_attributes_from_dict(dict, root):
+  [root.set(key, value) for key, value in dict.items()]
+
+
+def add_elements_from_dict(dict, root):
+  for element_name in dict:
+    text = dict[element_name].get(constants.TEXT, None)
+    attributes = dict[element_name].get(constants.ATTRIBUTES, None)
+    element = add_xml_element(root, element_name, text, attributes)
+    if constants.CHILD in dict[element_name]:
+      add_elements_from_dict(dict[element_name][constants.CHILD], element)
+
+
+def add_xml_element(root, element_name, text=None, attributes=None):
+  if is_vif_element(element_name):
+    element = ET.SubElement(root, get_vif_element_name(element_name))
+    if text:
+      element.text = str(text)
+    if attributes:
+      add_attributes_from_dict(attributes, element)
+    return element
+  return root
+
+
+def remove_special_characters(string):
+  return ''.join(filter(str.isalnum, string))
+
+
+def is_simple_data(data):
+  if isinstance(data, (str, int, bool)):
+    return True
+  return False
+
+
+def pdo_type(pdo_value):
+  return pdo_value >> 30
+
+
+def parse_sink_pdos(xml_mem, sink_pdos):
+  add_xml_element(xml_mem, constants.NUM_SINK_PDOS, None,
+                  {constants.VALUE: str(len(sink_pdos))})
+  add_xml_element(xml_mem, constants.EPR_SUPPORTED_AS_SINK,
+                  attributes={constants.VALUE: "false"})
+  add_xml_element(xml_mem, constants.NO_USB_SUSPEND_MAY_BE_SET,
```

Comment:
Should this value be gathered from parsing the 5V fixed sink cap?  Same for higher capability

Reply:
Oh ok, I assigned them to a boolean value directly based on how it's done in CrOS Genvif assuming it will work for current cases. I will check with Sam regarding it and get back to this comment. Thank you

Reply:
I updated how 'Higher Capability' and 'Fast Role Swap required USB Type-C' are set and getting value from Sink PDO. But didn't change anything for NO_USB_SUSPEND_MAY_BE_SET as it seems to be a member of Request Data Object instead of PDO, the plan for this I have in mind is, set it as device tree member and get the value from the DT in next revisions. Thank you